client/state/posts/counts/actions.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	POST_COUNTS_RECEIVE,
+	POST_COUNTS_REQUEST,
+	POST_COUNTS_REQUEST_SUCCESS,
+	POST_COUNTS_REQUEST_FAILURE
+} from 'state/action-types';
+
+/**
+ * Returns an action object signalling that post counts have been received for
+ * the site and post type.
+ *
+ * @param  {Number} siteId   Site ID
+ * @param  {String} postType Post type
+ * @param  {Object} counts   Mapping of post status to count
+ * @return {Object}          Action object
+ */
+export function receivePostCounts( siteId, postType, counts ) {
+	return {
+		type: POST_COUNTS_RECEIVE,
+		siteId,
+		postType,
+		counts
+	};
+}
+
+/**
+ * Returns an action thunk, dispatching progress of a request to retrieve post
+ * counts for a site and post type.
+ *
+ * @param  {Number}   siteId   Site ID
+ * @param  {String}   postType Post type
+ * @return {Function}          Action thunk
+ */
+export function requestPostCounts( siteId, postType ) {
```

Comment:
>Though not used in calypso, just noting that the endpoint does support passing in an `author(int)` to query the API for post-counts for a specific author [#](https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/post-counts/%24post_type/)

I might see how this could be useful for the case of filtering "my" vs. "everyone" posts, though not sure there's a huge performance win for querying only the author type, and would complicate the implementation quite a bit more.

Should a need arise in the future, it wouldn't be a stretch to imagine adding an additional argument or separate action creator for querying post counts specific to one author.